sendgrid_test.go

```diff
@@ -81,6 +81,53 @@ func TestGetRequestSubuser(t *testing.T) {
 	ShouldHaveHeaders(&request, t)
 }
 
+func Test_test_set_residency_eu(t *testing.T) {
```

Comment:
we should try to follow the `Go` convention of naming test functions and proper casing. You can refer to the other existing tests for example, but generally, it should be `Test<FunctionNameBeingTested>`

Reply:
Right, will correct the casing and naming of the test cases

Reply:
I think for different scenarios in a single function to be tested, we can add them with an underscore like the test cases like [TestRequestRetry_rateLimit](https://github.com/sendgrid/sendgrid-go/blob/main/sendgrid_test.go#L105) and [TestRequestRetry_rateLimit_noHeader](https://github.com/sendgrid/sendgrid-go/blob/main/sendgrid_test.go#L124).
Similarly, I think we can also have test cases named - TestSetDataResidency_eu and TestSetDataResidency_global. Will that be okay?